src/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -145,5 +145,10 @@ private string GetPropertyName(in ReadStackFrame frame)
 
             return propertyName;
         }
+
+        /// <summary>
+        /// Bytes consumed in the current loop
+        /// </summary>
+        public long BytesConsumed;
```

Comment:
Does this have to be on the `ReadStack` itself? Why not just a local outside the above reader loop?

Reply:
ReadStack is the state, seemed like a good place to keep all of the sate.